app/models/authentication.rb

```diff
@@ -55,6 +55,22 @@ def self.new(*args, &block)
     "invalid"     => 3,
   ).freeze
 
+  # Builds a case statement that case be used in a sql ORDER BY.
+  #
+  # Generated SQL looks like:
+  #
+  #   CASE
+  #   WHEN LOWER(status) = ''      THEN -1
+  #   WHEN LOWER(status) = 'valid' THEN 0
+  #   ...
+  #   ELSE -1
+  #
+  STATUS_SEVERITY_AREL = Arel::Nodes::Case.new.tap do |arel_case|
+    STATUS_SEVERITY.each do |value, sort_weight|
+      arel_case.when(arel_table[:status].lower.eq(value)).then(sort_weight)
```

Comment:
wow - turns out `arel_table[:status]` does not access the database.

who knew?